app.rb

```diff
@@ -0,0 +1,132 @@
+require './book'
+require './classroom'
+require './teacher'
+require './student'
+require './person'
+require './rentals'
+
+class App
+  attr_accessor :people, :rentals, :books
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts '#=> No books are available yet' if books.empty?
+    books.each do |book|
+      puts "#=> Title: #{book.title} Author: #{book.author}"
+    end
+    puts people.length
+  end
+
+  def list_all_people
+    puts '#=> No person is available yet' if people.empty?
+    people.each do |person|
+      puts "#=> [#{person.class}] ID: #{person.id} Name: #{person.name} Age: #{person.age}"
+    end
+    puts people.length
+  end
+
+  def create_teacher
+    print "Teacher's specialization: "
+    specialization = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    people << teacher
+    puts ''
+    puts '#=> Teacher created successfully'
+  end
+
+  def create_student
+    print 'Age: '
+    age = Integer(gets.chomp)
+    print 'Name: '
+    name = gets.chomp
+    print 'Parent permission? [y/n]: '
+    parent_permission = gets.chomp.downcase
+
+    case parent_permission
+    when 'n'
+      people << Student.new(nil, age, name, parent_permission: false)
+    when 'y'
+      people << Student.new(nil, age, name, parent_permission: true)
+    else
+      '#=> Invalid option, please try again'
+    end
+
```

Comment:
- Just like the previous reviewer suggested, I am still having errors when creating a person, a reason for this in your `Nameable.rb` file. Please remove the `initialize` block in the `nameable.rb` file. Please fix this 🙏